rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize.cpp

```diff
@@ -54,10 +53,83 @@ void ResizeNode::create_node()
     vx_status status;
     if((status = vxGetStatus((vx_reference)_node)) != VX_SUCCESS)
         THROW("Adding the resize (vxExtrppNode_ResizebatchPD) node failed: "+ TOSTR(status))
+}
 
+void ResizeNode::update_node() {
+    std::vector<uint32_t> src_h_dims, src_w_dims;
+    src_w_dims = _inputs[0]->info().get_roi_width_vec();
+    src_h_dims = _inputs[0]->info().get_roi_height_vec();
+    for (unsigned i = 0; i < _batch_size; i++) {
+        _src_width = src_w_dims[i];
+        _src_height = src_h_dims[i];
+        _dst_width = _out_width;
+        _dst_height = _out_height;
+        adjust_out_roi_size();
+        _dst_width = std::min(_dst_width, _outputs[0]->info().width());
+        _dst_height = std::min(_dst_height, _outputs[0]->info().height_single());
+        _dst_roi_width_vec.push_back(_dst_width);
+        _dst_roi_height_vec.push_back(_dst_height);
+    }
+    vx_status width_status, height_status;
+    width_status = vxCopyArrayRange((vx_array)_dst_roi_width, 0, _batch_size, sizeof(vx_uint32), _dst_roi_width_vec.data(), VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST);
+    height_status = vxCopyArrayRange((vx_array)_dst_roi_height, 0, _batch_size, sizeof(vx_uint32), _dst_roi_height_vec.data(), VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST);
+    if(width_status != 0 || height_status != 0)
+        WRN("ERROR: vxCopyArrayRange _dst_roi_width or _dst_roi_height failed " + TOSTR(width_status) + "  " + TOSTR(height_status));
+    _outputs[0]->update_image_roi(_dst_roi_width_vec, _dst_roi_height_vec);
+    _dst_roi_width_vec.clear();
+    _dst_roi_height_vec.clear();
 }
 
-void ResizeNode::update_node()
-{
-  
-}
\ No newline at end of file
+void ResizeNode::init(unsigned dest_width, unsigned dest_height, RocalResizeScalingMode scaling_mode,
+                      std::vector<unsigned> max_size, RocalResizeInterpolationType interpolation_type) {
+    _interpolation_type = (int)interpolation_type;
+    _scaling_mode = scaling_mode;
+    _out_width = dest_width;
+    _out_height = dest_height;
+    if(max_size.size() > 0) {
+        _max_width = max_size[0];
+        _max_height = max_size[1];
+    }
+}
+
+void ResizeNode::adjust_out_roi_size() {
+    bool has_max_size = (_max_width | _max_height) > 0;
+
+    if (_scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_STRETCH) {
+        if (_dst_width == 0) _dst_width = _src_width;
+        if (_dst_height == 0) _dst_height = _src_height;
+
+        if (has_max_size) {
+            if (_max_width != 0) _dst_width = std::min(_dst_width, _max_width);
+            if (_max_height != 0) _dst_height = std::min(_dst_height, _max_height);
+        }
+    } else if (_scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_DEFAULT) {
+        if (_dst_width == 0 && _dst_height != 0) {  // Only height is passed
```

Comment:
if you use logical statements instead, compiler generates better code than conditional statements
for e.g., if (!_dst_width & _dst_height)

Reply:
Done